src/main/java/edu/ucsd/sbrg/escher/gui/EscherConverterUI.java

```diff
@@ -341,22 +341,37 @@ protected File doInBackground() throws Exception {
   @Override
   public File saveFileAs() {
     String dir = SBPreferences.getPreferencesFor(GUIOptions.class).get(GUIOptions.SAVE_DIR);
-    JFileChooser fc = GUITools.createJFileChooser(dir, false, false, JFileChooser.FILES_ONLY,
-      SBFileFilter.createJSONFileFilter(),
-      SBFileFilter.createSBGNFileFilter(),
-      SBFileFilter.createSBMLFileFilterL3V1());
+    
+    // make sure the extension filter corresponds to the preferred output format
+    String format = SBPreferences.getPreferencesFor(EscherOptions.class).get(EscherOptions.FORMAT);
+    GeneralFileFilter filter = null;
+    switch (format){
+    case "SBGN": filter = SBFileFilter.createSBGNFileFilter(); break;
+    case "JSON": filter = SBFileFilter.createJSONFileFilter(); break;
+    default: filter = SBFileFilter.createSBMLFileFilterL3V1(); break;
+    }
+    JFileChooser fc = GUITools.createJFileChooser(dir, false, false, JFileChooser.FILES_ONLY, filter);
     File savedFile = null;
     if (fc.showSaveDialog(this) == JFileChooser.APPROVE_OPTION) {
-      savedFile = checkFile(fc.getSelectedFile());
+        savedFile = checkFile(fc.getSelectedFile());
       if ((savedFile != null) && (savedFile.getParentFile() != null)) {
         // Do conversion
         EscherMapDisplay display = (EscherMapDisplay) tabbedPane.getSelectedComponent();
         EscherMap map = display.getOpenedFile().getDocument();
+        
+        String filename = fc.getSelectedFile().getAbsolutePath().toString();
+        
         if (!SBFileFilter.isJSONFile(fc.getSelectedFile())) {
           EscherConverterWorker<?> converter;
           if (fc.getFileFilter().getDescription().toUpperCase().contains("SBGN")) {
             converter = new EscherConverterWorker<Sbgn>(map, Sbgn.class,
                 SBPreferences.getPreferencesFor(EscherOptions.class).toProperties());
+            
+            // make sure that pathname ends with .sbgn or .xml
+            if (!filename .endsWith(".sbgn") && !filename.endsWith(".xml"))
```

Comment:
When checking file names, it would be better to match patterns in a case-insensitive way. For instance, we can first convert to lower case or use a regular expression. On some file systems `.SbGn` is different from `.sbgN` or `.sbgn`, etc.